### PR TITLE
Attempt at fixing #2603 and superseding #2545

### PR DESCRIPTION
**Pull request by ozh**
This all boils down to redirecting to $_SERVER['REQUEST_URI'] after successful login and using less $_SERVER variables ('HTTP_HOST', 'HOSTNAME', 'SERVERNAME' or whatnot)

### CODE REVIEW

File: tests/includes/utils.php [hunk 1]
```diff
@@ -66,14 +66,17 @@ static function change_it( $var ) {
...
 }
- 
+
 /**
 * var_dump() for Unit Tests
 *
 * @since 0.1
 */
-function ut_var_dump( $what ) {
+function ut_var_dump( ...$what ) {
```
Comment (LeoColomb) [inline tests/includes/utils.php]: Wow, is it compatible with 5.6?

File: tests/includes/utils.php [hunk 2]
```diff
@@ -66,14 +66,17 @@ static function change_it( $var ) {
...
     ob_start();
+    if( count($what) == 1 ) {
```
Comment (LeoColomb) [inline tests/includes/utils.php]:
```suggestion
    if( count($what) === 1 ) {
```

File: includes/functions.php [hunk 3]
```diff
@@ -1998,13 +1998,29 @@ function yourls_get_request($yourls_site = false, $uri = false) {
...
+ * @param string $normal     Optional, the standard scheme (defaults to 'http://')
+ * @param string $ssl        Optional, the SSL scheme (defaults to 'https://')
+ * @return string            the modified URL, if applicable
  */
 function yourls_match_current_protocol( $url, $normal = 'http://', $ssl = 'https://' ) {
-	if( yourls_is_ssl() )
-		$url = str_replace( $normal, $ssl, $url );
-	return yourls_apply_filter( 'match_current_protocol', $url );
+    // we're only doing something if we're currently serving through SSL and the input URL begins with 'http://' or 'https://'
+    if( yourls_is_ssl() and in_array( yourls_get_protocol($url), array('http://', 'https://') ) ) {
```
Comment (LeoColomb) [inline includes/functions.php]:
```suggestion
    if( yourls_is_ssl() && in_array( yourls_get_protocol($url), array('http://', 'https://') ) ) {
```

File: tests/includes/utils.php (hunk 1 above)
Comment (LeoColomb) [reply tests/includes/utils.php]: Yes, it is: https://www.php.net/manual/en/functions.arguments.php#functions.variable-arg-list

**Comment by pcolmer** [comment]
@ozh I've tested this on our Docker installation and it fixes the issues I was encountering. Thank you.

**Comment by ozh** [comment]
Merging. Thanks @hangxingliu  and @pcolmer for insightful feedbacks which has led to this PR

**Pull request merged**